backend/public-graph/graph/resolver.go

```diff
@@ -2654,8 +2655,8 @@ func (r *Resolver) ProcessPayload(ctx context.Context, sessionSecureID string, e
 		util.Tag("numberOfErrors", len(errors)),
 		util.Tag("numberOfEvents", len(events.Events)),
 	}
-	// sample-in payload with many events
-	if len(events.Events) > 1_000 {
+	// sample-in 1/1000 of ProcessPayload calls
+	if env.IsProduction() && isIngestedBySample(ctx, "", 1./1000) {
```

Comment:
should this be based on the ctx traceID?

could also do this at a higher level, like at the kafka worker processing for all messages, or all pushpayload messages

Reply:
ah yes it should probably use the trace id, switching this over